vm/src/builtins/enumerate.rs

```diff
@@ -41,11 +41,9 @@ impl SlotConstructor for PyEnumerate {
     type Args = EnumerateArgs;
 
     fn py_new(cls: PyTypeRef, args: Self::Args, vm: &VirtualMachine) -> PyResult {
-        let counter = match args.start {
-            OptionalArg::Present(start) => start.as_bigint().clone(),
-            OptionalArg::Missing => BigInt::zero(),
-        };
-
+        let counter = args
+            .start
+            .map_or(BigInt::zero(), |start| start.as_bigint().clone());
```

Comment:
```suggestion
            .map_or_else(|| BigInt::zero(), |start| start.as_bigint().clone());
```
To make equivalent code, `map_or_else` fits better

Reply:
ah yes, good catch!